src/AspNet.Security.OAuth.Ebay/EbayAuthenticationHandler.cs

```diff
@@ -0,0 +1,146 @@
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace AspNet.Security.OAuth.Ebay
+{
+    public class EbayAuthenticationHandler : OAuthHandler<EbayAuthenticationOptions>
+    {
+        public EbayAuthenticationHandler(
+            IOptionsMonitor<EbayAuthenticationOptions> options,
+            ILoggerFactory logger,
+            UrlEncoder encoder,
+            ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity, [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Get, EbayAuthenticationDefaults.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            using var response = await this.Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, this.Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync(this.Context.RequestAborted));
+
+                throw new HttpRequestException("An error occurred while retrieving the user profile.");
+            }
+
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(this.Context.RequestAborted));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, this.Context, this.Scheme, this.Options, this.Backchannel, tokens, payload.RootElement);
+            context.RunClaimActions(payload.RootElement);
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal!, context.Properties, this.Scheme.Name);
+        }
+
+        protected override string BuildChallengeUrl([NotNull] AuthenticationProperties properties, [NotNull] string redirectUri)
+        {
+            var parameters = new Dictionary<string, string>
+            {
+                ["client_id"] = this.Options.ClientId,
+                ["redirect_uri"] = this.Options.RuName!,
+                ["response_type"] = "code",
+                ["scope"] = this.FormatScope(this.Options.Scope.Distinct().Select(s => NormalizeScope(s)))
```

Comment:
This should be achieved by overriding `FormatScope()` and having the implementation for the normalization there. Then this assignment just becomes `FormatScope(Options.Scope)`.